GitHubRepositories.ps1

```diff
@@ -1607,3 +1607,530 @@ function Move-GitHubRepositoryOwnership
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
+        None
+
+    .OUTPUTS
+        PSCustomObject
+
+    .EXAMPLE
+        Get-GitHubRepositoryVulnerabilityAlert -OwnerName Microsoft -RepositoryName PowerShellForGitHub
+
+        Retrieves the status of vulnerability alerts for the PowerShellForGithub repository.
+    .EXAMPLE
+        Get-GitHubRepositoryVulnerabilityAlert -Uri https://github.com/PowerShell/PowerShellForGitHub
+
+        Retrieves the status of vulnerability alerts for the PowerShellForGithub repository.
+#>
+    [CmdletBinding(
+        PositionalBinding = $false,
+        DefaultParameterSetName='Elements')]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            Position = 1,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog -Invocation $MyInvocation
+
+    $elements = Resolve-RepositoryElements -BoundParameters $PSBoundParameters
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $params = @{
+        UriFragment = "repos/$OwnerName/$RepositoryName/vulnerability-alerts"
+        Description =  "Getting Vulnerability Alerts status for $RepositoryName"
+        AcceptHeader = $script:dorianAcceptHeader
+        Method = 'Get'
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+        NoStatus = (Resolve-ParameterWithDefaultConfigurationValue -BoundParameters $PSBoundParameters `
+            -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    try
+    {
+        Invoke-GHRestMethod @params | Out-Null
+        $result = $true
+    }
+    catch
```

Comment:
following the pattern of the other `Test-*` methods, I think it's fine to just have the `catch` return `$false` and not `throw` anything, which would greatly simplify this part of the code.

Reply:
Not sure this is a good pattern. If a repository that doesn't exist is passed to the function, it would return $false rather than throw, which is misleading.

Reply:
In general, GitHub's API return back a 404 when:
  * The target object doesn't exist
  * The user doesn't have permission to the target object

They've already made it such that we can't guarantee a fully correct answer to certain queries.

Their documentation is confusing relative to what you've coded.  Per the documentation, receiving a 204 _No Content_ indicates that they are enabled, and a 404 indicates that they're not.  Their documentation doesn't guarantee that there will be a message included with the response (which your logic appears to be dependent on).

Reply:
As for the pattern, I think of `Test-*` methods to be similar to the `TryGet*` .NET pattern.  It should never throw.  I'm not 100% sure that that's standard PowerShell practice, but it feels like it should be.

Reply:
Pass an invalid `Path` to `Test-ModuleManifest` or `Test-PSSessionConfigurationFile` from the `Microsoft.PowerShell.Core` standard module or pass an invalid `CatalogFilePath` to `Test-FileCatalog` from the `Microsoft.PowerShell.Security` standard module. All will raise an exception.

Reply:
Quote from https://developer.github.com/v3/

> Resources may also send custom validation errors (where code is custom). Custom errors will always have a message field describing the error, and most errors will also include a documentation_url field pointing to some content that might help you resolve the error.

Reply:
I will concede this point.  Continue as previously submitted.